health-probe-proxy/Dockerfile

```diff
@@ -14,6 +14,6 @@
 
 # syntax=docker/dockerfile:1
 
-FROM gcr.io/distroless/static@sha256:ce46866b3a5170db3b49364900fb3168dc0833dfb46c26da5c77f22abb01d8c3
```

Comment:
Remvoed the sha, considering health-probe-proxy is comparatively stable, and we relies on dependent bot to bump the image to the latest. 
https://github.com/kubernetes-sigs/cloud-provider-azure/pull/6538/files